src/Couchbase.Lite.Shared/API/Sync/Replicator.cs

```diff
@@ -746,6 +752,14 @@ private C4Error SetupC4Replicator()
                 else
             #endif
                     _repl = Native.c4repl_new(Config.Database.c4db, addr, dbNameStr, _nativeParams.C4Params, &localErr);
+
+                if (_documentEndedUpdateEventsCount > 0) {
+                    C4Error err = new C4Error();
+                    if (!Native.c4repl_setProgressLevel(_repl, C4ReplicatorProgressLevel.PerDocument, &err) || err.code > 0) {
+                        WriteLog.To.Sync.W(Tag, "Failed set progress level to PerDocument", err);
+                    }
+                }
```

Comment:
I don't remember the nuance here, but why is this logic here instead of when adding the change listener.  What if you add a change listener after starting?

Reply:
c4 replicator would be null if I call setProgressLevel in the AddChangeListener. That is why I move the call down here after the c4 replicator is created.

Reply:
That doesn't account for the situation where a change listener is added after start though.  I think it is needed there too with a null check.